packages/base/core/src/Collection/index.ts

```diff
@@ -34,6 +34,7 @@ export interface CollectionOptions<T extends BaseItem<I>, I, U = T> {
   indices?: IndexProvider<T, I>[],
   enableDebugMode?: boolean,
   fieldTracking?: boolean,
+  primaryGenerator?: () => string,
```

Comment:
what about passing the existing item? 

```js
primaryGenerator?: (item: T) => string,
```

My use case is:

```js
primaryGenerator: (item) => typeid(item.type).toString()
```

Reply:
I like the idea of passing the existing item!
The return type also need to be changed to generic `I` as this represents the data type of ids in the collection